src/jack.py

```diff
@@ -1328,6 +1383,141 @@ def get_ports(self, name_pattern='', is_audio=False, is_midi=False,
             _lib.jack_free)
         return self._port_list_from_pointers(names)
 
+    def set_property(self, subject, key, value, type=''):
+        """Set a metadata property on *subject*.
+
+        Parameters
+        ----------
+        subject : int or str
+            The subject (UUID) to set the property on.
+            UUIDs can be obtained with `Client.uuid`, `Port.uuid` and
+            `Client.get_uuid_for_client_name()`.
+        key : str
+            The key (URI) of the property.  Some predefined keys are
+            available as ``jack.METADATA_*`` module constants.
+        value : str or bytes
+            The value of the property.
+        type : str, optional
+            The type of the property, either a MIME type or URI.
+            If *type* is empty, the *value* is assumed to be a UTF-8
+            encoded string (``'text/plain'``).
+
+            Example values:
+
+            - ``'image/png;base64'`` (base64 encoded PNG image)
+            - ``'http://www.w3.org/2001/XMLSchema#int'`` (integer)
+
+            Official types are preferred, but clients may use any
+            syntactically valid MIME type (which start with a type and
+            slash, like ``'text/...'``).  If a URI type is used, it must
+            be a complete absolute URI (which start with a scheme and
+            colon, like ``'http:'``).
+
+        See Also
+        --------
+        get_property
+        get_properties
+        get_all_properties
+        remove_property
+        remove_properties
+        remove_all_properties
+        set_property_change_callback
+
+        """
+        subject = _uuid_parse(subject)
+        if isinstance(value, str):
```

Comment:
This is problematic on Python 2. The `str` type is essentially the same as `bytes` on Python 3 there. It's value should not be encoded, it already is. `unicode` instances, on the other hand, should be encoded. One solution is to use `isinstance(s, six.text_type)` or have something like this at the top of your module:

    if bytes is str:
        text_type = unicode
    else:
        text_type = str

(But I'm not sure whether the latter works on Python 2.6.)

Reply:
I've used this in the past:

```python
try:
    _unicode = unicode  # doesn't exist in Python 3.x
except NameError:
    _unicode = str
```

I think adding a dependency on `six` is overkill here.